packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js

```diff
@@ -108,74 +130,111 @@ function getMenuItem(item, index, getComponent) {
 };
  <ActionDropdown {...props} />
  */
-function ActionDropdown(props) {
-	const {
-		bsStyle,
-		hideLabel,
-		icon,
-		items,
-		label,
-		link,
-		onSelect,
-		tooltipPlacement,
-		tooltipLabel,
-		getComponent,
-		components,
-		className,
-		...rest
-	} = props;
-
-	const Renderers = Inject.getAll(getComponent, { MenuItem, DropdownButton });
-	const injected = Inject.all(getComponent, components, InjectDropdownMenuItem);
-	const title = (
-		<span className="tc-dropdown-button-title">
-			{icon ? <Icon name={icon} /> : null}
-			{hideLabel ? null : <span className="tc-dropdown-button-title-label">{label}</span>}
-		</span>
-	);
-	const style = link ? 'link' : bsStyle;
+class ActionDropdown extends React.Component {
+	constructor(props) {
+		super(props);
+		this.onToggle = this.onToggle.bind(this);
+		this.state = {
+			dropup: props.dropup,
+		};
+	}
+
+	onToggle(isOpen, event) {
+		if (!isOpen) {
+			this.setState({ dropup: this.props.dropup });
+			return;
+		}
 
-	function onItemSelect(object, event) {
-		if (onSelect) {
-			onSelect(event, object);
+		const dropdownTrigger = getDropdownToggleFromInner(event.target);
+		const dropdownMenu = dropdownTrigger.nextSibling;
+		const dropdownContainer = getDropdownContainer(dropdownTrigger);
+
+		if (dropdownContainer) {
+			const dropdownRect = dropdownMenu.getBoundingClientRect();
+			const containerRect = dropdownContainer.getBoundingClientRect();
+
+			let dropup = this.state.dropup;
```

Comment:
this state update should be done as a function since it rely on previous dropup state